joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/AstForPrimitivesCreator.scala

```diff
@@ -204,12 +204,13 @@ trait AstForPrimitivesCreator(implicit withSchemaValidation: ValidationMode) {
     argName: Option[String],
     annotations: Seq[KtAnnotationEntry] = Seq()
   )(implicit typeInfoProvider: TypeInfoProvider): Ast = {
-    val (fullName, signature) = typeInfoProvider.fullNameWithSignature(expr, ("", "")) // TODO: fix the fallback names
-    val typeFullName          = registerType(typeInfoProvider.expressionType(expr, TypeConstants.javaLangObject))
+    val typeFullName = registerType(typeInfoProvider.expressionType(expr, TypeConstants.javaLangObject))
+    val fullName     = "<operator>.class"
```

Comment:
wouldn't a `TypeRef` be a great fit for `::class`?

Reply:
No, `::class` can give you the class object for some like a class name `MyClass::class` but also for any object like `someObject::class`. In the later case there is no static information about a type to populate a TypeRef. A TypeRef is used for the `MyClass` identifier though.

Reply:
I see, makes sense now.